skbio/stats/_subsample.py

```diff
@@ -147,19 +149,21 @@ def bin_f(x):
             yield (bin_, item)
 
 
-def subsample_counts(counts, n, replace=False):
+def subsample_counts(counts, n, replace=False, seed=None):
     """Randomly subsample from a vector of counts, with or without replacement.
 
     Parameters
     ----------
     counts : 1-D array_like
-        Vector of counts (integers) to randomly subsample from.
+        Vector of counts (integers or floats) to randomly subsample from.
     n : int
         Number of items to subsample from `counts`. Must be less than or equal
         to the sum of `counts`.
     replace : bool, optional
         If ``True``, subsample with replacement. If ``False`` (the default),
         subsample without replacement.
+    seed : int, optional
```

Comment:
This `seed` may be (see [here](https://github.com/scikit-bio/scikit-bio/blob/0a9fc577ebf7ce894e3b3b67e5b3a5cb732076c5/skbio/stats/composition.py#L1880) for an example:

```
    seed : int or np.random.Generator, optional
        A user-provided random seed or random generator instance.
```

Such that a user can either provide an integer, or a pre-constructed rng. In th latter scenario, a user can do `rng = np.random.default_rng(42)` in the beginning of an analysis, and use it throughout the entire analysis.